build/teamcity-testlogicrace.sh

```diff
@@ -8,23 +8,27 @@ maybe_ccache
 
 mkdir -p artifacts
 
-run_json_test build/builder.sh \
+TESTTIMEOUT=1h
+
+run_json_test build/builder.sh env \
```

Comment:
what's this `env` for? I don't think this does anything. If anything, I think you want `env TESTTIMEOUT="${TESTTIMEOUT}" make ...` but it's not needed. I'd say just remove this here and below.

Rest looks good!